sumpy/tools.py

```diff
@@ -116,7 +116,8 @@
 # {{{ multi_index helpers
 
 def add_mi(mi1: Sequence[int], mi2: Sequence[int]) -> tuple[int, ...]:
-    return tuple([mi1i + mi2i for mi1i, mi2i in zip(mi1, mi2)])
+    # NOTE: these are used a lot and `tuple([])` is faster
+    return tuple([mi1i + mi2i for mi1i, mi2i in zip(mi1, mi2)])  # noqa: C409
```

Comment:
I recall there was a lot of multi-index juggling in here somewhere, so this keeps the faster comprehension version.